src/number_theory/modular.lean

```diff
@@ -323,10 +323,21 @@ by rw [←coe_re, coe_T_zpow_smul_eq, add_re, int_cast_re, coe_re]
 lemma im_T_zpow_smul (n : ℤ) : ((T^n) • z).im = z.im :=
 by rw [←coe_im, coe_T_zpow_smul_eq, add_im, int_cast_im, add_zero, coe_im]
 
-lemma re_T_smul : (T • z).re = z.re + 1 := by simpa using re_T_zpow_smul z 1
-lemma im_T_smul : (T • z).im = z.im := by simpa using im_T_zpow_smul z 1
-lemma re_T_inv_smul : (T⁻¹ • z).re = z.re - 1 := by simpa using re_T_zpow_smul z (-1)
-lemma im_T_inv_smul : (T⁻¹ • z).im = z.im := by simpa using im_T_zpow_smul z (-1)
+lemma re_T_smul : (T • z).re = z.re + 1 := by simp only [modular_T_smul, vadd_re, add_comm]
+
+lemma im_T_smul : (T • z).im = z.im := by simp only [modular_T_smul, vadd_im]
+
+lemma re_T_inv_smul : (T⁻¹ • z).re = z.re - 1 :=
+by simpa only [upper_half_plane.denom.equations._eqn_1, int.coe_cast_ring_hom,
+  algebra_map.coe_one, matrix.special_linear_group.coe_GL_pos_coe_GL_coe_matrix,
+  matrix.map_apply,zpow_one, zpow_neg, upper_half_plane.num.equations._eqn_1,
+  upper_half_plane.re_smul, int.cast_neg,
+  matrix.special_linear_group.coe_matrix_coe, upper_half_plane.sl_moeb,
+  complex.of_real_int_cast, coe_coe,
+  matrix.special_linear_group.coe_inv] using re_T_zpow_smul z (-1)
```

Comment:
This `simpa only` looks far too long to me; surely you only need that `T^(-1) = T⁻¹` and `z.re - 1 = z.re + (-1)`?

Reply:
That's a copy-paste of the output of "by simpa? using foo`; I tried using `squeeze_simpa` but it times out on my machine.

To put it bluntly: I really don't want to get involved in rewriting `number_theory/modular.lean`. It has absolutely nothing to do with the aims of this ticket, and I have no knowledge whatsoever of the layout of that file or the design decisions behind it. 

The only reason why I made any changes to it at all is that I added a simp lemma elsewhere which apparently broke one of the proofs here, for a totally stupid reason: apparently a proof of the form `by simpa using foo` can break if you add new simp-lemmas (it seems that if the new lemma allows the simplifier to close the goal without using `foo`, then it responds `match failed: no goals`). See [this workflow run](https://github.com/leanprover-community/mathlib/actions/runs/4372621106/jobs/7649766641). Since the documentation explicitly recommends `simpa using` as being more robust against changes in the simp set than plain `simp`, I'd consider this a bug in `simpa` -- it certainly should not be seen as a *problem* that the goal is easy to close! If it weren't for this tactic bug I'd have no reason to be editing `number_theory/modular.lean` at all.

Reply:
Discussion on Zulip suggests that others do not agree with my belief that this is a tactic bug; so I am going to back off and avoid editing these files at all.